rma/models/rma.py

```diff
@@ -0,0 +1,904 @@
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tests import Form
+from odoo.tools import html2plaintext
+import odoo.addons.decimal_precision as dp
+from odoo.addons.stock.models.stock_move import PROCUREMENT_PRIORITIES
+
+
+class Rma(models.Model):
+    _name = "rma"
+    _description = "RMA"
+    _order = "priority,date desc"
+    _inherit = ["mail.thread", "portal.mixin", "mail.activity.mixin"]
+
+    # General fields
+    sent = fields.Boolean()
+    name = fields.Char(
+        string='Name',
+        index=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        copy=False,
+        default=lambda self: _('New'),
+    )
+    origin = fields.Char(
+        string='Source Document',
+        states={
+            'locked': [('readonly', True)],
+            'cancelled': [('readonly', True)],
+        },
+        help="Reference of the document that generated this RMA.",
+    )
+    date = fields.Datetime(
+        default=lambda self: fields.Datetime.now(),
+        Index=True,
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    deadline = fields.Date(
+        states={
+            'locked': [('readonly', True)],
+            'cancelled': [('readonly', True)],
+        },
+    )
+    user_id = fields.Many2one(
+        comodel_name="res.users",
+        string="Responsible",
+        track_visibility="always",
+        default=lambda self: self.env.user,
+        states={
+            'locked': [('readonly', True)],
+            'cancelled': [('readonly', True)],
+        },
+    )
+    team_id = fields.Many2one(
+        comodel_name="rma.team",
+        string="RMA team",
+        index=True,
+        states={
+            'locked': [('readonly', True)],
+            'cancelled': [('readonly', True)],
+        },
+    )
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        default=lambda self: self.env.user.company_id,
+        states={
+            'locked': [('readonly', True)],
+            'cancelled': [('readonly', True)],
+        },
+    )
+    partner_id = fields.Many2one(
+        string="Customer",
+        comodel_name="res.partner",
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        index=True,
+        track_visibility='always'
+    )
+    partner_invoice_id = fields.Many2one(
+        string="Invoice Address",
+        comodel_name="res.partner",
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        domain=[('customer', '=', True)],
+        help="Refund address for current RMA."
+    )
+    picking_id = fields.Many2one(
+        comodel_name="stock.picking",
+        string="Origin Delivery",
+        domain="["
+               "    ('state', '=', 'done'),"
+               "    ('picking_type_id.code', '=', 'outgoing'),"
+               "    ('partner_id', '=', partner_id),"
```

Comment:
Maybe a little too restrictive. As partner_id is not required here, no deliveries at all are shown by default.

Reply:
Is it wanted ?

Reply:
Well, maybe it's an overcomplication (it would require additional onchanges, etc) as RMA's born from the customer anyway. The forget my comment here @ernestotejeda :)

Reply:
Yeah, let's keep it this way for now.